libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1464,23 +1472,27 @@ string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
 					default {
 						sstore(array, add(data, 2))
 						let slot, offset := <indexAccess>(array, oldLen)
-						<storeValue>(slot, offset, value)
+						<storeValue>(slot, offset <values>)
 					}
 				<!isByteArray>
 					let oldLen := sload(array)
 					if iszero(lt(oldLen, <maxArrayLength>)) { <panic>() }
 					sstore(array, add(oldLen, 1))
 					let slot, offset := <indexAccess>(array, oldLen)
-					<storeValue>(slot, offset, value)
+					<storeValue>(slot, offset <values>)
 				</isByteArray>
 			})")
 			("functionName", functionName)
+			("values", _fromType->sizeOnStack() == 0 ? "" : ", " + suffixedVariableNameList("value", 0, _fromType->sizeOnStack()))
 			("panic", panicFunction(PanicCode::ResourceError))
 			("extractByteArrayLength", _type.isByteArray() ? extractByteArrayLengthFunction() : "")
 			("dataAreaFunction", arrayDataAreaFunction(_type))
 			("isByteArray", _type.isByteArray())
 			("indexAccess", storageArrayIndexAccessFunction(_type))
-			("storeValue", updateStorageValueFunction(*_type.baseType(), *_type.baseType()))
+			(
+				"storeValue",
+				updateStorageValueFunction(_fromType ? *_fromType : *_type.baseType(), *_type.baseType())
```

Comment:
```suggestion
				updateStorageValueFunction(*_fromType)
```
Somehow I think this comment got lost earier - this was the second one I meant.
Looking at the beginning of the function ``_fromType`` is never null here and we already use that for ``values`` anyways.

Reply:
But also rather minor, so fine with merging without.

Reply:
Will change.

Reply:
I think you forgot to dereference :-).

Reply:
Sorry - fixed.